mobile/src/lib/cozy-helper.js

```diff
@@ -83,10 +83,10 @@ export function refreshFolder (dispatch, getState) {
 }
 
 export const onError = (dispatch, getState) => (err) => {
-  if (err.message === clientRevokedMsg) {
+  if (err.message === clientRevokedMsg || err.error === 'code=400, message=Invalid JWT token') {
     console.warn(`Your device is no more connected to your server: ${getState().mobile.settings.serverUrl}`)
     dispatch(revokeClient())
-  } else if (err.message === 'ETIMEDOUT' || err.message === 'getCheckpoint rejected with ') {
+  } else if (err.message === 'ETIMEDOUT') {
```

Comment:
It appears that pouchDB replication encapsulates network error in an error with message `getCheckpoint rejected with`. For instance when your token is not correct, cozy-stack responds with a `Invalid JWT Token` translated by cozy-client-js to `Client has been revoked`. On the other hand, the pouchDB replication encapsulates the `Invalid JWT Token` in an error with `getCheckpoint rejected with` as the message.
So it is not very accurate as it catches others errors than `timeout` errors and I prefer to remove this and use `err.error` to see what's the real error message is.